### PR TITLE
mpich: add 3.4

### PR DESCRIPTION
**Pull request by michaelkuhn**
This also switches the default device to ch4 and the default netmod to ofi (tcp does not exist for ch4).

cc @raffenet @yfguo

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]
Waiting on maintainer approval

**Review by alalazo — APPROVED** [review]

**Comment by haampie** [comment]
Ping @raffenet / @yfguo, are you OK with this PR?

**Comment by yfguo** [comment]
The PR looks good to me.

**Comment by raffenet** [comment]
ch4:ucx also supports TCP, though it is not a configuration we test regularly in our CI. Also, I just pushed out a 3.4.1 bug-fix update this afternoon... 😬

**Comment by michaelkuhn** [comment]
> ch4:ucx also supports TCP, though it is not a configuration we test regularly in our CI.

We currently have a conflict for ch4 with tcp, should we remove that?

> Also, I just pushed out a 3.4.1 bug-fix update this afternoon...

#21302 :smile:

**Comment by raffenet** [comment]
> > ch4:ucx also supports TCP, though it is not a configuration we test regularly in our CI.

No, sorry. What I mean to say is that both ch4:ofi and ch4:ucx support MPI communication over TCP/IP. There is no native ch4:tcp netmod, but you can run MPI applications over TCP using either configuration.

ch4:ofi is a fine default for many cases, especially non-Linux platforms. For Linux systems with Infiniband, we recommend ch4:ucx.

**Comment by michaelkuhn** [comment]
> No, sorry. What I mean to say is that both ch4:ofi and ch4:ucx support MPI communication over TCP/IP. There is no native ch4:tcp netmod, but you can run MPI applications over TCP using either configuration.

Now that I reread your comment, I see the ucx reference, sorry.

> ch4:ofi is a fine default for many cases, especially non-Linux platforms. For Linux systems with Infiniband, we recommend ch4:ucx.

So, should we keep ofi as the default then? Probably makes sense, since we also support non-Linux systems.

**Comment by raffenet** [comment]
> > No, sorry. What I mean to say is that both ch4:ofi and ch4:ucx support MPI communication over TCP/IP. There is no native ch4:tcp netmod, but you can run MPI applications over TCP using either configuration.
> 
> Now that I reread your comment, I see the ucx reference, sorry.
> 
> > ch4:ofi is a fine default for many cases, especially non-Linux platforms. For Linux systems with Infiniband, we recommend ch4:ucx.
> 
> So, should we keep ofi as the default then? Probably makes sense, since we also support non-Linux systems.

Yes, I think that makes sense for now.

**Pull request merged**
